src/job.ts

```diff
@@ -5,24 +5,39 @@ import {Nullable} from './nullable'
 
 const NO_JOB = '--no-job--'
 
+export enum JobState {
```

Comment:
Yes, I think the `Queue` class would be much cleaner if we start using Jobs instead of dealing with CommittedMessages. but we can refactor that before the next queue feature.